cmd/rm-main.go

```diff
@@ -121,17 +133,54 @@ func checkRmSyntax(ctx *cli.Context) {
 	isForce := ctx.Bool("force")
 	isRecursive := ctx.Bool("recursive")
 	isStdin := ctx.Bool("stdin")
+	isDangerous := ctx.Bool("dangerous")
+	isNamespaceRemoval := false
+	for _, url := range ctx.Args() {
+		// check if url represents just the namespace without bucket or object prefix specified
+		// in the url
+		aliasedURL := filepath.ToSlash(url)
+		alias, path := url2Alias(aliasedURL)
+		splits := splitStr(aliasedURL, "/", 3)
+		bucket := splits[1]
+		_, _, err := url2Stat(aliasedURL)
+
+		// check if bucket name is empty
+		isBucketEmpty := false
+		if _, ok := err.ToGoError().(BucketNameEmpty); ok {
+			isBucketEmpty = true
+		}
+		// validate if prefix exists
+		if !isBucketEmpty && err != nil && !isURLPrefixExists(aliasedURL, false) {
+			fatalIf(err.Trace(aliasedURL), "Failed to remove `"+aliasedURL+"`.")
+		}
+		// validate extraneus delimiters such as play// which won't be caught by previous check
+		if path != "" && bucket == "" {
+			fatalIf(err.Trace(aliasedURL), "Failed to remove `"+aliasedURL+"`.")
 
+		}
+		if alias != "" && path == "" && isBucketEmpty {
+			isNamespaceRemoval = true
+			break
```

Comment:
The whole checker logic is confusing to me.  In `rm`  we should not check for bucket or object which is completely abstracted by client interface.  I see below logic is good enough

```go
url = filepath.ToSlash(url)
if !isAliasURLDir(url) {
    alias, path := url2Alias(url)
    isNamespaceRemoval = (path == "")
}
```

Reply:
without the check on lines 157-158, errors like mc rm play// wont be caught. Earlier it was being caught in s3 api because bucket name was not permitted to be empty. Now by allowing namespace removal, bucket name can be empty and there is no way to validate cases like multiple consecutive / delimiters being entered as input. Lines 160-163 can be replaced by the logic you proposed.

Reply:
I see `url = filepath.ToSlash(url)` is wrong, you should use `url = filepath.Clean(url)`

Reply:
done.